x/cdp/keeper/auctions.go

```diff
@@ -0,0 +1,189 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/cdp/types"
+)
+
+type partialDeposit struct {
+	Depositor sdk.AccAddress
+	Amount    sdk.Coins
+	DebtShare sdk.Int
+}
+
+func newPartialDeposit(depositor sdk.AccAddress, amount sdk.Coins, ds sdk.Int) partialDeposit {
+	return partialDeposit{
+		Depositor: depositor,
+		Amount:    amount,
+		DebtShare: ds,
+	}
+}
+
+type partialDeposits []partialDeposit
+
+func (pd partialDeposits) SumCollateral() (sum sdk.Int) {
+	sum = sdk.ZeroInt()
+	for _, d := range pd {
+		sum = sum.Add(d.Amount[0].Amount)
+	}
+	return
+}
+
+func (pd partialDeposits) SumDebt() (sum sdk.Int) {
+	sum = sdk.ZeroInt()
+	for _, d := range pd {
+		sum = sum.Add(d.DebtShare)
+	}
+	return
+}
+
+// AuctionCollateral creates auctions from the input deposits which attempt to raise the corresponding amount of debt
+func (k Keeper) AuctionCollateral(ctx sdk.Context, deposits types.Deposits, debt sdk.Int, bidDenom string) {
+	auctionSize := k.getAuctionSize(ctx, deposits[0].Amount[0].Denom)
+	partialAuctionDeposits := partialDeposits{}
+	totalCollateral := deposits.SumCollateral()
+	for deposits.SumCollateral().GTE(auctionSize) {
+		for i, dep := range deposits {
+			// create auctions from individual deposits that are larger than the auction size
+			k.CreateAuctionsFromDeposit(ctx, &dep, &debt, &totalCollateral, auctionSize, bidDenom)
+			if !dep.Amount[0].Amount.IsZero() {
```

Comment:
Found these loops a bit hard to understand, comments might help explain why things are happening